### PR TITLE
Me/add maven atlassian

### PR DESCRIPTION
**Pull request by mellisio**

### CODE REVIEW

File: ingestors/maven.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		return "@every 1h"
+	case "hortonworks":
+		return "@every 1h"
+	case "springLibsRelease":
+		return "@every 6h"
+	default:
+		return "@every 10h"
+	}
+}
+func (ingestor *MavenIngestor) getLibrariesName() string {
```
Comment (jsonperl) [inline ingestors/maven.go:50]: We've got a `Namer` interface now that just has a `Name()` function that probably makes sense to use here.

File: ingestors/maven.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	case "mavenCentral":
+		return "maven_mavencentral"
+	case "atlassian":
+		return "maven_atlassian"
+	case "hortonworks":
+		return "maven_hortonworks"
+	case "springLibsRelease":
+		return "maven_springlibs"
+	default:
+		return "maven"
```
Comment (jsonperl) [inline ingestors/maven.go:61]: This should probably just be an error yea? Maybe explode on initialization if not in a known list.

**Comment by jsonperl** [comment]
In rubyland I'd probably slap a hash up top with all the info and use that to raise an exception if we don't know what the name is. Wonder if something like that is a viable pattern in Go.

```
repos = {"mavenCentral": "@every 12h"}...
```

File: ingestors/maven.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	case "mavenCentral":
+		return "maven_mavencentral"
+	case "atlassian":
+		return "maven_atlassian"
+	case "hortonworks":
+		return "maven_hortonworks"
+	case "springLibsRelease":
+		return "maven_springlibs"
+	default:
+		log.Fatal(fmt.Sprintf("Unknown maven ingestor name: %s", ingestor.FeedName))
```
Comment (jsonperl) [inline ingestors/maven.go:61]: logrus has a Fatalf, I think there's no need for a return here then?

Comment (jsonperl) [reply ingestors/maven.go:61]: I think the more important place to fail would be initialization though.

File: ingestors/maven.go [hunk 4]
```diff
@@ -13,9 +13,13 @@ type MavenIngestor struct {
...
 
 func NewMaven(feedName string) *MavenIngestor {
-	return &MavenIngestor{
+	maven := &MavenIngestor{
 		FeedName: feedName,
 	}
+	if maven.Name() == "" {
+		log.Fatalf("Unknown maven ingestor name: %s", feedName)
+	}
+	return maven
```
Comment (mellisio) [inline ingestors/maven.go:22]: not entirely sure about this style. How would you handle it? Sadly no const maps https://groups.google.com/g/golang-nuts/c/xUaDMciyB5Q?pli=1 as far as I can tell https://stackoverflow.com/questions/18342195/how-to-declare-a-constant-map-in-golang

Comment (jsonperl) [reply ingestors/maven.go:22]: Hmm would something like this be nice?
```
type MavenRepository string

const(
    MavenCentral MavenRepository = "mavenCentral"
    Atlassian = "atlassian"
    Hortonworks = "hortonworks"
    SpringLibsRelease = "springLibsRelease"
)
```

Then we could initialize via
`func NewMaven(repository MavenRespository) *MavenIngestor` passing like `ingestors.MavenCentral`

Comment (tiegz) [reply ingestors/maven.go:22]: Yeah, bummer about const maps. I think the switch statement makes the most sense (altho not sure how that performs compared to a function that returns a map 🤷 )

**Review by tiegz — APPROVED** [review]

File: ingestors/maven.go (hunk 4 above)
Comment (jsonperl) [reply ingestors/maven.go:22]: The main issue that we should tackle is exploding on init if something goofy is added. So whatever route we go, let's do that.

File: ingestors/maven_central.go [hunk 5]
```diff
@@ -1,40 +0,0 @@
...
-
-func NewMavenCentral() *MavenCentral {
-	return &MavenCentral{}
-}
-
-func (ingestor *MavenCentral) Schedule() string {
-	return mavenCentralSchedule
-}
-
-func (ingestor *MavenCentral) TTL() time.Duration {
```
Comment (jsonperl) [inline ingestors/maven_central.go:27]: We lost the customized TTL in the new code

File: ingestors/maven.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	results, err := mp.GetPackages()
+	if err != nil {
+		log.WithFields(log.Fields{"ingestor": ingestor.Name(), "error": err}).Error()
+		return results
+	}
+	ingestor.LatestRun = time.Now()
+	return results
+}
+func (ingestor *MavenIngestor) TTL() time.Duration {
+	return ttl
```
Comment (jsonperl) [inline ingestors/maven.go:42]: We're going to want this to be different for each one no?

File: ingestors/maven.go (hunk 4 above)
Comment (jsonperl) [reply ingestors/maven.go:24]: > Hmm would something like this be nice?
> 
> ```
> type MavenRepository string
> 
> const(
>     MavenCentral MavenRepository = "mavenCentral"
>     Atlassian = "atlassian"
>     Hortonworks = "hortonworks"
>     SpringLibsRelease = "springLibsRelease"
> )
> ```
> 
> Then we could initialize via
> `func NewMaven(repository MavenRespository) *MavenIngestor` passing like `ingestors.MavenCentral`

Were you not into this idea? I think it's safer / easier to use.

File: ingestors/maven.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+func (ingestor *MavenIngestor) getSchedule() string {
+	switch ingestor.FeedName {
+	case "mavenCentral":
+		return "@every 12h"
```
Comment (jsonperl) [inline ingestors/maven.go:48]: Is there any harm in just running these all hourly for everything if we're deduping with the appropriate ttl?

File: ingestors/maven.go (hunk 6 above)
Comment (jsonperl) [reply ingestors/maven.go:42]: I guess whats the harm except some more redis storage

**Review by jsonperl — APPROVED** [review]
This works, and I'd like to get it deployed so we're not missing packages, so I'm gonna merge it.

**Pull request merged**